data-prepper-plugins/opensearch/src/main/java/com/amazon/dataprepper/plugins/sink/opensearch/index/IndexManagerFactory.java

```diff
@@ -18,4 +18,36 @@ public final IndexManager getIndexManager(final IndexType indexType,
         }
     }
 
+    private class DefaultIndexManager extends IndexManager {
```

Comment:
Please make these `private static class` instead of `private class`. See my note in your old PR: https://github.com/opensearch-project/data-prepper/pull/361#discussion_r727118927.

Also, here is another relevant quote from the Java documentation, regarding inner classes (without the static modifier):

> As with instance methods and variables, an inner class is associated with an instance of its enclosing class and has direct access to that object's methods and fields. 

This is not something that we need or want in this scenario.

Reply:
Make sense. Will add static